### ISSUE TITLE
parallelize operation

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandisets#90]
with growing size of the archive, going serially through the files becomes suboptimal -- we really do not saturate the bandwidth.  Two choices
- an ideal way would be to parallelize download/addition of new files at git-annex level. After #84 RFing I expect us to just use `annex addurl` and `annex registerurl` (filed relevant https://github.com/dandi/dandisets/issues/89).  There is  `annex addurl --batch -J JOBS --json` -- needs checking if it would internally handle them nicely/out of order. Code would then need to proceed with the url/path just added to `registerurl --batch` to populate more urls... Might work quite nicely 
- "cheap" solution -- parallelize across dandisets at 
```
        for d in self.get_dandisets(dandiset_ids, exclude=exclude):
            dsdir = self.target_path / d.identifier
```
but that would be suboptimal since then "large" dandisets would still be taking awhile whenever others might be already done.

So I think we should implement the former approach.  Not sure if should be a part of a solution for #84 or done after. The first thing would be to check how `addurl --batch --jobs JOBS --json` would behave
